test/nodes/test_extractor.py

```diff
@@ -109,6 +117,21 @@ def test_extractor_indexing(document_store):
     assert "Haystack" in meta["entity_words"]
 
 
+@pytest.mark.integration
+def test_extractor_doc_query():
```

Comment:
Why aren't we using the `ner_node` fixture in this test?

Reply:
Since I wanted to test the NER node in a query pipeline with the `flatten_entities_in_meta_data=True` option, which isn't `False` by default in the fixture test. The `flatten_entities_in_meta_data` is needed for metadata filtering to work out of the box with `OpenSearch`

Reply:
I think we can still use the `ner_node` fixture and just set `ner_node.flatten_entities_in_meta_data = True` at the beginning of the test.

By the way, using `OpenSearchDocumentStore`, you don't need to flatten the metadata fields to be able to use them for filtering, see also [OpenSearch Documentation on Nested fields](https://opensearch.org/docs/1.3/opensearch/supported-field-types/nested/). For example, if you want to filter your Documents for the `PER` label, you can use the following filter expression: `filters={"entities.entity_group": "PER"}`.

Reply:
> I think we can still use the ner_node fixture and just set ner_node.flatten_entities_in_meta_data = True at the beginning of the test.

In general I'm not a huge fan of this in case we ever change the `scope` of the pytest fixture (e.g. `pytest.fixture(scope="module")`) and the fixture is then only loaded once for the whole `.py` file. If the state of the node is then changed in any of the tests then this might cause the other tests to fail. I can set `ner_node.flatten_entities_in_meta_data = True` this, but then I'll also manually add a `ner_node.flatten_entities_in_meta_data = False` at the end of the test to undo the changing the state of the node. 

Thanks for the extra knowledge on the `OpenSearchDocumentStore`! I wasn't aware of this, I guess this is only a requirement currently on the dC side, which is where I was given this restriction initially.

Reply:
I ended up reusing the ner_node fixture and removing the `flatten_entities_in_meta_data = True` to prevent changing the state of the node. Should be good to go!